enhancements/ingress/lb-subnet-selection-aws.md

```diff
@@ -0,0 +1,608 @@
+perpetuates the pattern of ensuring that `status.endpointPublishingStrategy` reflects the
+_actual_ value (in our case, the annotation value).
+
+#### Install Time Configuration
+
+TODO
+
+### Workflow Description
+
+#### Setting IngressController Subnets during Installation Workflow
+
+Setting the default subnets for all IngressController during installation (Day 1):
+
+TODO: Workflow for install time configuration
+
+#### Creating a Custom IngressController with Subnets after Installation Workflow
+
+Creating an IngressController while specifying the subnets for the LoadBalancer-type
+service after installation:
+
+1. A cluster admin creates an IngressController with
+  `spec.endpointPublishingStrategy.loadBalancer.providerParameters.aws.subnets` specified.
+2. The ingress operator creates the LoadBalancer-type Service and sets the
+   `service.beta.kubernetes.io/aws-load-balancer-subnets` annotation with the subnets
+   provided by the user.
+
+#### Updating an existing IngressController with new Subnets Workflow
+
+Updating an existing IngressController with new subnets requires the service
+to be deleted and is disruptive:
+
+1. A cluster admin edits an IngressController's
+   `spec.endpointPublishingStrategy.loadBalancer.providerParameters.aws.subnets` field.
+2. The Ingress Operator sets the `LoadBalancerProgressing` condition to `Status: True`
+   on the IngressController and provides a message indicating to the cluster admin
+   that they must delete the service.
+3. The cluster admin deletes the LoadBalancer-type service indicated in the
+   `LoadBalancerProgressing` condition message.
+4. The Ingress Operator recreates the LoadBalancer-type service and configures the
+   `service.beta.kubernetes.io/aws-load-balancer-subnets` on the LoadBalancer-type
+   service, thus creating the load balancer with the desired subnets specified by the
+   cluster admin.
+
+#### Unmanaged Subnet Annotation Migration Workflow
+
+Migrating an unmanaged `service.beta.kubernetes.io/aws-load-balancer-subnets`
+service annotation to `spec.endpointPublishingStrategy.loadBalancer.providerParameters.aws.subnets`
+after upgrading to 4.y doesn't require a cluster admin to delete the LoadBalancer-type
+service:
+
+1. Cluster admin confirms that initially `service.beta.kubernetes.io/aws-load-balancer-subnets`
+   is set on the LoadBalancer-type service managed by the IngressController.
+2. The cluster admin upgrades OpenShift to v4.y and the service annotation is not
+   changed or removed.
+3. After upgrading, the IngressController will emit a `LoadBalancerProgressing` condition
+   with `Status: True` because the spec's `Subnets` (an empty slice `[]`) does not equal
+   the current annotation.
+4. Instead of recreating the LoadBalancer-type service, the cluster admin will set
+   `spec.endpointPublishingStrategy.loadBalancer.providerParameters.aws.subnets` to the
+   current value of the `service.beta.kubernetes.io/aws-load-balancer-subnets` service.
```

Comment:
Is it true that they could either delete the service or fix the controller subnets?  If so, I would keep it simple and say:
```suggestion
4. The cluster admin can either delete the service as usual, OR they may manually set
   `spec.endpointPublishingStrategy.loadBalancer.providerParameters.aws.subnets` to the
   current value of the `service.beta.kubernetes.io/aws-load-balancer-subnets` service.  This is the only time this manual operation is recommended.
```

Reply:
Ok, I misunderstood because of the "Instead of recreating the ..service", which definitely shouldn't be done.  Maybe just:
```suggestion
4. In this case,  the cluster admin must directly update
   `spec.endpointPublishingStrategy.loadBalancer.providerParameters.aws.subnets` to the
   current value of the `service.beta.kubernetes.io/aws-load-balancer-subnets` service.
```

Reply:
Done.